### PR TITLE
add aten.add.int lowering in TorchToStd

### PR DESCRIPTION
**Pull request by dan-garvey**
I'm not super familiar with LIT testing, so if I did something weird there please lmk

### CODE REVIEW

File: lib/Dialect/Torch/Transforms/RefineTypes.cpp [hunk 1]
```diff
@@ -575,6 +580,13 @@ static ResultTypeState updateResultTypeState(ValueKnowledge *tensor,
   return new_state;
 }
 
+static Type getPromotedResultType(ArrayRef<Type> scalarTypes) {
```
Comment (cathyzhyi) [inline lib/Dialect/Torch/Transforms/RefineTypes.cpp:583]: Shall we add testing for scalar_scalar case here https://github.com/llvm/mlir-npcomp/blob/56c6e3676bbe1571dab0c7cc1fb5080846819438/test/Dialect/Torch/promote-types.mlir

File: lib/Dialect/Torch/Transforms/RefineTypes.cpp [hunk 2]
```diff
@@ -1074,6 +1086,13 @@ ChangeResult TypeAnalyzer::visitScalarToTensorConversionOp(OpTy op) {
   return getLatticeElement(op.getResult()).join(knowledge);
 }
 
+template <typename OpTy> ChangeResult TypeAnalyzer::visitScalarOp(OpTy op) {
```
Comment (cathyzhyi) [inline lib/Dialect/Torch/Transforms/RefineTypes.cpp:1089]: nit: change method name to `visitBinaryScalarOp`

File: test/Conversion/TorchToStd/basic.mlir [hunk 3]
```diff
@@ -74,3 +74,14 @@ func @torch.constant.int() -> !torch.int {
   %int1 = torch.constant.int 1
   return %int1 : !torch.int
 }
+
+// CHECK-LABEL:  func @torch.aten.add.int(%arg0: !torch.int, %arg1: !torch.int) -> !torch.int {
+// CHECK:    %[[LHS_I64:.*]] = torch_c.to_i64 %[[LHS]]
```
Comment (cathyzhyi) [inline test/Conversion/TorchToStd/basic.mlir:79]: Are the CHECK tags generated from `generate-test-checks.py`? The space between CHECK tag and the content to check against are not the same as the previous tests.
```
// CHECK:           return %[[FLOAT]] : !torch.float
vs
// CHECK:    return %[[INT:.*]] : !torch.int
```

File: lib/Conversion/TorchToStd/TorchToStd.cpp [hunk 4]
```diff
@@ -44,6 +44,19 @@ class ConvertAtenDimOp : public OpConversionPattern<AtenDimOp> {
 };
 } // namespace
 
+namespace {
```
Comment (cathyzhyi) [inline lib/Conversion/TorchToStd/TorchToStd.cpp:47]: We also need an e2e test for this.

**Review by cathyzhyi — CHANGES_REQUESTED** [review]

**Comment by pashu123** [comment]
Now that e2e testing with elemental-types as return types are supported, you should create an e2e test. You can pass the arguments as a zero-d tensor and convert it to scalar using int(x) fn, then perform the required addition.

File: test/Conversion/TorchToStd/basic.mlir (hunk 3 above)
Comment (dan-garvey) [reply test/Conversion/TorchToStd/basic.mlir:79]: Busted. I wrote them manually, thanks.

File: lib/Dialect/Torch/Transforms/RefineTypes.cpp (hunk 1 above)
Comment (cathyzhyi) [reply lib/Dialect/Torch/Transforms/RefineTypes.cpp:583]: Oh, actually, we can do that in a different PR. We would need the `Torch_AtenAddOp` for testing this. The `Torch_AtenAddIntOp` would always have operands with the same type so won't really test the logic. Could you open an issue saying we need to add a testing for scalar-scalar type promotion?

File: e2e_testing/torchscript/scalar.py [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+        ([], torch.long, True),
+        ([], torch.long, True),
+    ])
+    def forward(self, lhs, rhs):
+        return int(lhs)+int(rhs)
+
+
+@register_test_case(module_factory=lambda: AddIntModule())
+def AddIntModule_basic(module, tu: TestUtils):
+    module.forward(torch.tensor(3, dtype=torch.long), torch.tensor(3, dtype=torch.long))
```
Comment (cathyzhyi) [inline e2e_testing/torchscript/scalar.py:29]: prefer to use  something like `torch.randint(-100, 100,()).` to generated random testing data.

File: e2e_testing/torchscript/scalar.py [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+
+class AddIntModule(torch.nn.Module):
+    def __init__(self):
+        super().__init__()
+
+    @export
+    @annotate_args([
+        None,
+        ([], torch.long, True),
+        ([], torch.long, True),
```
Comment (cathyzhyi) [inline e2e_testing/torchscript/scalar.py:21]: For consistency, use `torch.int64` for dtype.

**Review by cathyzhyi — CHANGES_REQUESTED** [review]

**Review by cathyzhyi — APPROVED** [review]

**Pull request merged**